docs/source/cpp/streaming_execution.rst

```diff
@@ -175,9 +175,607 @@ their completion::
+or stddev is returned.
+
+Example::
+
+    // ddof: 
+    // skip_nulss: If true (the default), null values are ignored. 
+    //////Otherwise, if any value is null, emit null.
+    // min_count: If less than this many non-null values are observed, emit null.
+    arrow::compute::VarianceOptions variance_option(/*ddof/*1, 
+                                                    /*skip_nulls*/true, 
+                                                    /*min_count*/3);
+
+5. `QuantileOptions`
+
+:class:`arrow::compute::QuantileOptions` This option controls the Quantile kernel behavior. 
+By default, returns the median value. There is an interpolation method to use when quantile 
+lies between two data points. The provided options for interpolation are; `LINEAE`, `LOWER`, `HIGHER`,
+`NEAREST` and `MIDPOINT`.
+
+Example::
+
+    // q: quantile must be between 0 and 1 inclusive 
+    ////// (scalar value or a std::vector as input)
+    // interpolation: one of `LINEAER`, `LOWER`, 'HIGHER', 
+    ////// `NEAREST`, `MIDPOINT`
+    // skip_nulls: If true (the default), null values are ignored. Otherwise, 
+    ////// if any value is null, emit null.
+    // min_count: If less than this many non-null values are observed, emit null.
+    arrow::compute::QuantileOptions quantile_options(/*q*/0.50, 
+      /*interpolation*/cp::QuantileOptions::Interpolation::LINEAR, 
+      /*skip_nulls*/true, 
+      /*min_count*/3);
+
+6. `TDigestOptions`
+
+`arrow::compute::TDigestOptions` option controls TDigest approximate quantile kernel behavior.
+By default, returns the median value.
+
+Example::
+
+    // q: quantile must be between 0 and 1 inclusive
+    // delta: compression parameter, default 100
+    // buffer_size: input buffer size, default 500
+    // skip_nulls: if true (the default), null values are ignored. Otherwise, if any value is null,
+    ////// emit null.
+    // min_count: If less than this many non-null values are observed, emit null.
+    arrow::compute::TDigestOptions tdigest_option(/*q*/0.5, 
+      /*delta*/200, 
+      /*buffer_size*/600, 
+      /*skip_nulls*/true, 
+      /*min_count*/5);
+
+7. IndexOptions
+
+:class:`arrow::compute::IndexOptions` This option controls Index kernel behavior. 
+This is used to find the index of a particular scalar value. 
+
+Example::
+
+    arrow::compute::IndexOptions index_options(arrow::MakeScalar("1"));
+
+An example for creating an aggregate node::
+
+    arrow::compute::CountOptions options(arrow::compute::CountOptions::ONLY_VALID);
+
+    auto aggregate_options = arrow::compute::AggregateNodeOptions{
+        /*aggregates=*/{{"hash_count", &options}},
+        /*targets=*/{"a"},
+        /*names=*/{"count(a)"},
+        /*keys=*/{"b"}};
+
+    ARROW_ASSIGN_OR_RAISE(cp::ExecNode * aggregate,
+                              cp::MakeExecNode("aggregate", plan.get(), {source},
+                              aggregate_options));
+
+
+Scan-Node
```

Comment:
```suggestion
``ScanNode``
```

Reply:
But there is no entity like that in the source code, this was another unclear part. 
The definition set is not consistent. Some of them have a node and some have the 
functionality but a node is not exposed even internally. For instance "MakeScaneNode" 
is not a function exposed.

Reply:
Yup, I see your point. I think we should just refer to these as "scan", "filter", "project" etc. since users should never need to interact with the actual classes.

Reply:
Replaced the nodes with `scan`, `project`, etc operations and didn't explicitly discussed about nodes.